app/MMB/src/scenes/tripOverview/TimelineDeparture.js

```diff
@@ -31,24 +38,44 @@ function TimelineDeparture(props: Props) {
 
       <View style={styleSheet.row}>
         <View style={styleSheet.detailedInfo}>
-          <View style={styleSheet.row}>
-            <Text style={styleSheet.key}>
-              <Translation id="mmb.flight_overview.timeline.carrier" />
-            </Text>
-            <Text style={styleSheet.value}>
-              <Translation passThrough={` ${airlineName || ''}`} />
-            </Text>
-          </View>
+          <TimelineRow
+            icon={
+              <TextIcon
+                code="a"
+                style={[styleSheet.icon, styleSheet.carrierIcon]}
+              />
+            }
+            title={<Translation id="mmb.flight_overview.timeline.carrier" />}
+            value={<Translation passThrough={` ${airlineName || ''}`} />}
+          />
+
+          <TimelineRow
+            icon={<TextIcon code="R" style={styleSheet.icon} />}
+            title={<Translation id="mmb.flight_overview.timeline.flight_no" />}
+            value={<Translation passThrough={` ${flightNumber || ''}`} />}
+          />
 
-          <View style={styleSheet.row}>
-            {/* TODO: this looks weird when we need to work with flights and busses - how to fix? */}
-            <Text style={styleSheet.key}>
-              <Translation id="mmb.flight_overview.timeline.flight_no" />
-            </Text>
-            <Text style={styleSheet.value}>
-              <Translation passThrough={` ${flightNumber || ''}`} />
-            </Text>
-          </View>
+          <TimelineRow
+            icon={<TextIcon code="R" style={styleSheet.icon} />}
+            value={
+              <Translation passThrough={`${manufacturer} ${flightModel}`} />
+            }
+          />
+          {terminal !== '' && (
+            <TimelineRow
+              icon={<TextIcon code="*" style={styleSheet.icon} />}
+              value={
+                <Translation
+                  id="mmb.flight_overview.timeline.terminal"
+                  values={{
+                    terminal,
+                    city,
+                    iata,
+                  }}
+                />
+              }
+            />
+          )}
```

Comment:
There should also be one like that for the TimelineArrival_routeStop where the icon is the landing plane.

Reply:
It is no landing icon in the current app

Reply:
I will do this in another PR. There is still more missing parts here anyways.